### DISCUSSION TITLE
How to calculate the area value of the segmented object

### DISCUSSION THREAD

**Discussion by TodorovaA** [roboflow/supervision#1015] [Q&A]
Dear @SkalskiP ,

How can I calculate the area on the segments that are detected? And that to run on a video or in real-time.

Appreciate it if anyone has any ideas or insights.

Thank you

**Comment by rolson24** [comment]
Hi @ana111todorova,

What model are you using to get the segmentation masks? If you are using SAM, I believe the area is a value in the output array of dictionaries from SAM and you can get it by just doing `output_mask[mask_index]['Area']`.
If you are using Ultralytics yolo, I don't think they give you the area directly. But what you can do is get the masks into roboflow Detections. Then because they store the masks as numpy arrays where a 1 is masked off, and a zero is not masked, you can just do np.sum(mask) to get the area of the mask. Remember to first get the subarray for the detection you want the area of. 
For example if you want the area of the mask for detection 'i' do this:
```
current_mask = detections.mask[i]
area = np.sum(current_mask)
```

**Reply by TodorovaA** [reply]
Thank you
